Computer, Controller Software

JLefebvre55 opened the pull request:
Closes #130 

## Testing Documentation

All unit tests pass on hardware. All functionality displayed on hardware.

JLefebvre55 committed:
LED Class Improvements

led color enum, parsing to String

JLefebvre55 committed:
Testing Suite, ICSP Flashing

Tests: GPIO miniUART serial port functionality (loopback test), compilation and upload test (blink), revision check test
Arduino compile and flash utility function (utils)
Readme overhaul
Removed deprecated Arduino libraries (scale, nau7802, float)

JLefebvre55 committed:
More Sophisticated Sensors, Actuators

Also:
- More code formatting
- Folders for sensors, actuators
- .env template
- Updated control flow diagram
- Minor fixes

JLefebvre55 committed:
Arduino Communications Fixes

Fixed startup order of operations (Arduino Serial begin vs Post, initial instructions, etc.)
No more dynamic allocation
Added logging function to Spinner (pauses and restarts spinner)
Fixed batch publishing in local logging mode

JLefebvre55 committed:
Minor Changes

Arduino accepts empty instruction set
Fixed spinner log ui util
Better C++ styling (header order, file naming, #include protection naming)

JLefebvre55 committed:
File Cases

JLefebvre55 committed:
Package v0.0.9

JLefebvre55 committed:
Delete .message

JLefebvre55 committed:
Minor Fixes

JLefebvre55 committed:
Migrated to PlatformIO

* Migrated to PlatformIO for compilation
* Separated JSON messaging, instruction parsing into utility files
* Moved globals to header
+ PIO Testing Suite
* Updated README

JLefebvre55 committed:
Sensor Unit Tests

JLefebvre55 committed:
TypeScript Unit Testing

* Fixed PIO unit testing
* JSON messenger returns instead of printing to serial during unit tests
* main.cc ignored during unit tests
* Moved lib/ to src/
+ Setup jest
+ Utils unit tests
+ Arduino unit testing from TS
- Removed python tests, test readme

JLefebvre55 committed:
Fixed Unit Tests

* Fixed TS compiling for RPi Node version (12.22.x)
* No longer running test_instructions or test_messenger (awaiting fix)

JLefebvre55 committed:
Actuator Control

+ Direct Arduino instruction set command via GCP
* Fixed actuator target statefulness
- no more ignored tests
- no more "F" PSTR
* Fixed TS serial write error handling

JLefebvre55 committed:
GPIO Utils, Arduino Reset on Start

+ Utils for exporting, unexporting, writing to, and reading from GPIO pins via sysfs
+ Testing suite for new utils
* Arduino reset pin pulled high on start

JLefebvre55 committed:
Restructured Data Batches

* De-nested, from {"air-temperature":{batch:[...]},...} to {"air-temperature":[...],...}

JLefebvre55 committed:
Minor Fixes

+ Added Jest config via gitignore "exclude"
- Removed camera

JLefebvre55 committed:
Overhaul

* Simulated interfaces accessible only via unit testing
+ Util methods for compound .env checking, returning and loading to objects
* Moved "simulated" interfaces to be in the same file as the classes they simulate
+ PeaPod class
+ More+better variable description comments
* Refactored "arduino" to "controller"
- Removed instances of "PeaPod" in type names (redundant)
- Removed "findSerialPath", useless (should ALWAYS rely on .env)
* Controller serial timeout now forces chip reset rather than fail
* Moved away from "then", instead using await (better for exception handling)
* Followed Google TS style guide :)

JLefebvre55 committed:
Renaming, Removing Deprecated

* Renamed PeaPodOS-Arduino to microcontroller
- Removed checkArduino and corresponding unit test

JLefebvre55 committed:
Merge branch 'staging' into subsystems-automation

JLefebvre55 committed:
BoMs

JLefebvre55 committed:
Merge branch 'staging' into subsystems-automation

JLefebvre55 committed:
Minor Fix

Made intervals private fields

JLefebvre55 committed:
PeaPodOS v0.1.1

JLefebvre55 committed:
Fixed Reset

- Removed sysfs GPIO util functions
+ Implemeneted `onoff` package GPIO

JLefebvre55 committed:
Supply Pump, Solenoid Actuators

JLefebvre55 committed:
TEC Actuator Class

JLefebvre55 committed:
Idle State

JLefebvre55 committed:
Water Temperature, Pressure, Flow Sensor Classes

* namespaced instruction handler functions
* optimized main loop to speed up instruction handling

JLefebvre55 committed:
Microcontroller Optimization, UX, Interrupts

+ Added sensor and actuator name strings (progmem), toString methods
* Moved datasetup to progmem
* Moved globals to static class properties
* Replaced all String with c strings
+ Added InterruptHandler interface to allow hardware interrupts to modify object instance data using a small hack
* Modified variable storage properties where fitting (i.e. const char* const* as opposed to const char**)

JLefebvre55 committed:
Microcontroller Modular Unit Tests

JLefebvre55 committed:
Schedule Parsing, Control Systems

JLefebvre55 committed:
Minor Changes

Commit:
added linting and extensions for vscode

Commit:
pretty printed files

JLefebvre55 committed:
Docs WIP

JLefebvre55 committed:
Archived WIP

JLefebvre55 committed:
Clean Slate

JLefebvre55 committed:
Import Main Script

JLefebvre55 committed:
Revamped Main

JLefebvre55 committed:
Add I2CIP

dogkai012-coder commented:
My computer display card had a troubles  I don't want make troubles bit I had many troubles fd they say I can find a mult solve ways to avoid any situation any guys help me @@ black screen/ cpu 100 / fishing ／／ Trojan/ data steal ／ back door my anti virus can't counter them

JLefebvre55 committed:
Air Module w/ Tests

JLefebvre55 committed:
Watering, Lighting Modules w/ Tests

JLefebvre55 committed:
Added ADC

JLefebvre55 committed:
Minor Fixes

JLefebvre55 committed:
Modularized

JLefebvre55 committed:
Update workspace files and normalize line endings

JLefebvre55 committed:
Import I2CIP.js

JLefebvre55 committed:
Made Frontend Selectable

JLefebvre55 committed:
Bug Fixes

JLefebvre55 committed:
SerialPort Scanner

JLefebvre55 committed:
Module State HashTables & Listing, Configuration Handling

JLefebvre55 committed:
Minor Changes

JLefebvre55 committed:
Minor Fixes

JLefebvre55 committed:
State Table

JLefebvre55 committed:
Cleanup, Readme

JLefebvre55 committed:
Sorting State Table

JLefebvre55 committed:
State Editor

JLefebvre55 committed:
State Locking

JLefebvre55 committed:
Minor Fixes

JLefebvre55 committed:
Minor Changes

JLefebvre55 committed:
Firmware Flashing

JLefebvre55 committed:
Fixed FSM Bug

JLefebvre55 committed:
Camera

JLefebvre55 committed:
Minor Fixes

JLefebvre55 committed:
Docker, Fixed Types

JLefebvre55 committed:
Minor Fixes

JLefebvre55 committed:
Menu

JLefebvre55 committed:
New Event/Interval Scheduler w/ Calendar, DataGrid

refactored old scheduler to intervals

JLefebvre55 committed:
Minor Fixes

rebuild state tree on modal open
command instruction validation
fixed interval-start timeout
bumped up controller reset interval

JLefebvre55 committed:
Save & Load Schedules

JLefebvre55 committed:
Minor Fixes

JLefebvre55 committed:
Minor Fixes
